### ISSUE TITLE
sendEmbed and sendMessage not firing

### ISSUE THREAD

**Issue by valzargaming** [teamreflex/DiscordPHP#453]
**Environment**
- PHP Version:
    - 7.4.3
- DiscordPHP Version:
    - Latest master

**Describe the bug**
in the context of GUILD_MEMBER_REMOVE and GUILD_MEMBER_ADD, sendMessage and sendEmbed seems to randomly fail to execute and none of the callbacks ever trigger.

**To Reproduce**
Code snippet:
```php
if ($welcome_log_channel) {
	echo '[TEST]' . __FILE__ . ':' . __LINE__ . PHP_EOL;
	//Send the message, announcing the member's departure
	$welcome_log_channel->sendMessage("", false, $embed)->done(function ($r) {
		ob_flush();
		ob_start();
		var_dump($r);
		file_put_contents("result_dump.txt", ob_get_flush());
	}, function ($error) {
		ob_flush();
		ob_start();
		var_dump($error);
		file_put_contents("error_dump.txt", ob_get_flush());
	});
	return true;
}
```

**Expected behavior**
sendEmbed and sendMessage should fire and receive a callback, or at least emit an error as to why they failed to fire.

**Screenshots**
![image](https://user-images.githubusercontent.com/7202504/101288008-e7141180-37c1-11eb-9da9-30c2f4c5b014.png)
![image](https://user-images.githubusercontent.com/7202504/101288037-32c6bb00-37c2-11eb-83ca-c5fa16c38181.png)

**Additional context**
I am currently unable to reliably replicate this issue. The var_dump of the channels being used show no irregularities. Isuse occurs with both `sendEmbed->($embed`) and `sendMessage("", false, $embed)`

**Comment by valzargaming** [comment]
I have also confirmed this issue is happening with member->addRole() in the same context

**Comment by davidcole1340** [comment]
Could you add an HTTP logger?

```php
$logger = new Monolog('HTTPLogger');
$logger->pushHandler(new StreamHandler('php://stdout'));

$discord = new Discord([
    // ...
    'httpLogger' => $logger,
]);
```

**Comment by valzargaming** [comment]
```php
[2020-12-07T09:41:55.193021+00:00] HTTPLogger.DEBUG: REQ GET https://discord.com
/api/v6/gateway/bot queued [] []
[LOGIN] 07-12-2020 09:41:55
[2020-12-07T09:41:55.373928+00:00] HTTPLogger.DEBUG: REQ GET https://discord.com
/api/v6/gateway/bot successful [] []
[2020-12-07T09:41:55.374291+00:00] DiscordPHP.INFO: gateway retrieved and set {"
gateway":"wss://gateway.discord.gg/?v=6&encoding=json","session":{"total":1000,"
remaining":986,"reset_after":6615604,"max_concurrency":1}} []
[2020-12-07T09:41:55.374585+00:00] DiscordPHP.INFO: starting connection to webso
cket {"gateway":"wss://gateway.discord.gg/?v=6&encoding=json"} []
[2020-12-07T09:41:55.512393+00:00] DiscordPHP.INFO: websocket connection has bee
n created [] []
[2020-12-07T09:41:55.513844+00:00] DiscordPHP.INFO: received hello [] []
[2020-12-07T09:41:55.514130+00:00] DiscordPHP.INFO: heartbeat timer initilized {
"interval":41250.0} []
[2020-12-07T09:41:55.514301+00:00] DiscordPHP.INFO: identifying {"payload":{"op"
:2,"d":{"token":"[REDACTED]","p
roperties":{"$os":"WINNT","$browser":"DiscordBot (https://github.com/teamreflex/
DiscordPHP, v5.0.12)","$device":"DiscordBot (https://github.com/teamreflex/Disco
rdPHP, v5.0.12)","$referrer":"https://github.com/teamreflex/DiscordPHP","$referr
ing_domain":"https://github.com/teamreflex/DiscordPHP"},"compress":true}}} []
[2020-12-07T09:41:55.731971+00:00] HTTPLogger.DEBUG: REQ GET https://discord.com
/api/v6/oauth2/applications/@me queued [] []
[2020-12-07T09:41:55.732311+00:00] DiscordPHP.INFO: did not parse private channe
ls [] []
[2020-12-07T09:41:55.735752+00:00] DiscordPHP.INFO: stored guilds {"count":0,"un
available":52} []
[2020-12-07T09:41:56.200285+00:00] DiscordPHP.INFO: all guilds are now available
 {"count":52} []
[2020-12-07T09:41:56.200591+00:00] DiscordPHP.INFO: set up chunking, checking fo
r chunks every 5 seconds [] []
[2020-12-07T09:41:56.301668+00:00] DiscordPHP.INFO: client is ready [] []
[READY] Logged in as Palace Bot#9203 (662093882795753482)
[READY TIMESTAMP] 07-12-2020 09:41:56
[2020-12-07T09:41:56.314545+00:00] HTTPLogger.DEBUG: REQ GET https://discord.com
/api/v6/oauth2/applications/@me successful [] []
guildMemberAdd (468979034571931648)
[TEST]C:\Discord Bots\discord - Palace DiscordPHP\guildmemberadd-include.php:44
[TEST]C:\Discord Bots\discord - Palace DiscordPHP\guildmemberadd-include.php:62
[2020-12-07T09:42:46.282490+00:00] HTTPLogger.DEBUG: REQ POST https://discord.co
m/api/v6/channels/468979761390288896/messages queued [] []
[TEST]C:\Discord Bots\discord - Palace DiscordPHP\guildmemberadd-include.php:88
[2020-12-07T09:42:46.513620+00:00] HTTPLogger.DEBUG: REQ POST https://discord.co
m/api/v6/channels/679374168462065668/messages queued [] []
[2020-12-07T09:42:46.516747+00:00] HTTPLogger.DEBUG: REQ POST https://discord.co
m/api/v6/channels/468979761390288896/messages successful [] []
[2020-12-07T09:42:46.713884+00:00] HTTPLogger.DEBUG: REQ POST https://discord.co
m/api/v6/channels/679374168462065668/messages successful [] []
guildMemberRemove (468979034571931648)
[TEST]C:\Discord Bots\discord - Palace DiscordPHP\guildmemberremove-include.php:
71
[2020-12-07T09:42:50.773190+00:00] HTTPLogger.DEBUG: REQ POST https://discord.co
m/api/v6/channels/468979761390288896/messages queued [] []
[2020-12-07T09:42:50.898403+00:00] HTTPLogger.WARNING: REQ POST https://discord.
com/api/v6/channels/468979761390288896/messages failed: Discord\Exceptions\Disco
rdRequestFailedException: Bad Request in C:\Discord Bots\discord - Palace Discor
dPHP\vendor\team-reflex\discord-php\src\Discord\Http\Http.php:355 Stack trace: #
0 C:\Discord Bots\discord - Palace DiscordPHP\vendor\team-reflex\discord-php\src
\Discord\Http\Http.php(292): Discord\Http\Http->handleError(Object(RingCentral\P
sr7\Response)) #1 C:\Discord Bots\discord - Palace DiscordPHP\vendor\react\promi
se\src\FulfilledPromise.php(42): Discord\Http\Http->Discord\Http\{closure}(Objec
t(RingCentral\Psr7\Response)) #2 C:\Discord Bots\discord - Palace DiscordPHP\ven
dor\react\promise\src\Promise.php(66): React\Promise\FulfilledPromise->done(Obje
ct(Closure), Object(Closure)) #3 C:\Discord Bots\discord - Palace DiscordPHP\ven
dor\react\promise\src\Promise.php(168): React\Promise\Promise::React\Promise\{cl
osure}(Object(React\Promise\FulfilledPromise)) #4 C:\Discord Bots\discord - Pala
ce DiscordPHP\vendor\react\promise\src\Promise.php(231): React\Promise\Promise->
settle(Object(React\Promise\FulfilledPromise)) #5 C:\Discord Bots\discord - Pala
ce DiscordPHP\vendor\react\promise\src\Deferred.php(36): React\Promise\Promise::
React\Promise\{closure}(Object(RingCentral\Psr7\Response)) #6 C:\Discord Bots\di
scord - Palace DiscordPHP\vendor\react\http\src\Io\Transaction.php(90): React\Pr
omise\Deferred->resolve(Object(RingCentral\Psr7\Response)) #7 C:\Discord Bots\di
scord - Palace DiscordPHP\vendor\react\promise\src\FulfilledPromise.php(28): Rea
ct\Http\Io\Transaction->React\Http\Io\{closure}(Object(RingCentral\Psr7\Response
)) #8 C:\Discord Bots\discord - Palace DiscordPHP\vendor\react\promise\src\Promi
se.php(134): React\Promise\FulfilledPromise->then(Object(Closure), Object(Closur
e)) #9 C:\Discord Bots\discord - Palace DiscordPHP\vendor\react\promise\src\Prom
ise.php(168): React\Promise\Promise::React\Promise\{closure}(Object(React\Promis
e\FulfilledPromise)) #10 C:\Discord Bots\discord - Palace DiscordPHP\vendor\reac
t\promise\src\Promise.php(231): React\Promise\Promise->settle(Object(React\Promi
se\FulfilledPromise)) #11 C:\Discord Bots\discord - Palace DiscordPHP\vendor\rea
ct\promise\src\FulfilledPromise.php(42): React\Promise\Promise::React\Promise\{c
losure}(Object(RingCentral\Psr7\Response)) #12 C:\Discord Bots\discord - Palace
DiscordPHP\vendor\react\promise\src\Promise.php(135): React\Promise\FulfilledPro
mise->done(Object(Closure), Object(Closure), Object(Closure)) #13 C:\Discord Bot
s\discord - Palace DiscordPHP\vendor\react\promise\src\Promise.php(168): React\P
romise\Promise::React\Promise\{closure}(Object(React\Promise\FulfilledPromise))
#14 C:\Discord Bots\discord - Palace DiscordPHP\vendor\react\promise\src\Promise
.php(231): React\Promise\Promise->settle(Object(React\Promise\FulfilledPromise))
 #15 C:\Discord Bots\discord - Palace DiscordPHP\vendor\react\promise\src\Fulfil
ledPromise.php(42): React\Promise\Promise::React\Promise\{closure}(Object(RingCe
ntral\Psr7\Response)) #16 C:\Discord Bots\discord - Palace DiscordPHP\vendor\rea
ct\promise\src\Promise.php(66): React\Promise\FulfilledPromise->done(Object(Clos
ure), Object(Closure)) #17 C:\Discord Bots\discord - Palace DiscordPHP\vendor\re
act\promise\src\Promise.php(168): React\Promise\Promise::React\Promise\{closure}
(Object(React\Promise\FulfilledPromise)) #18 C:\Discord Bots\discord - Palace Di
scordPHP\vendor\react\promise\src\Promise.php(231): React\Promise\Promise->settl
e(Object(React\Promise\FulfilledPromise)) #19 C:\Discord Bots\discord - Palace D
iscordPHP\vendor\react\promise\src\FulfilledPromise.php(42): React\Promise\Promi
se::React\Promise\{closure}(Object(RingCentral\Psr7\Response)) #20 C:\Discord Bo
ts\discord - Palace DiscordPHP\vendor\react\promise\src\Promise.php(135): React\
Promise\FulfilledPromise->done(Object(Closure), Object(Closure), Object(Closure)
) #21 C:\Discord Bots\discord - Palace DiscordPHP\vendor\react\promise\src\Promi
se.php(168): React\Promise\Promise::React\Promise\{closure}(Object(React\Promise
\FulfilledPromise)) #22 C:\Discord Bots\discord - Palace DiscordPHP\vendor\react
\promise\src\Promise.php(231): React\Promise\Promise->settle(Object(React\Promis
e\FulfilledPromise)) #23 C:\Discord Bots\discord - Palace DiscordPHP\vendor\reac
t\promise\src\FulfilledPromise.php(42): React\Promise\Promise::React\Promise\{cl
osure}('{"embed": ["fie...') #24 C:\Discord Bots\discord - Palace DiscordPHP\ven
dor\react\promise\src\Promise.php(135): React\Promise\FulfilledPromise->done(Obj
ect(Closure), Object(Closure), Object(Closure)) #25 C:\Discord Bots\discord - Pa
lace DiscordPHP\vendor\react\promise\src\Promise.php(168): React\Promise\Promise
::React\Promise\{closure}(Object(React\Promise\FulfilledPromise)) #26 C:\Discord
 Bots\discord - Palace DiscordPHP\vendor\react\promise\src\Promise.php(231): Rea
ct\Promise\Promise->settle(Object(React\Promise\FulfilledPromise)) #27 C:\Discor
d Bots\discord - Palace DiscordPHP\vendor\react\promise-stream\src\functions.php
(75): React\Promise\Promise::React\Promise\{closure}('{"embed": ["fie...') #28 C
:\Discord Bots\discord - Palace DiscordPHP\vendor\evenement\evenement\src\Evenem
ent\EventEmitterTrait.php(123): React\Promise\Stream\{closure}() #29 C:\Discord
Bots\discord - Palace DiscordPHP\vendor\react\http\src\Io\ReadableBodyStream.php
(50): Evenement\EventEmitter->emit('close') #30 C:\Discord Bots\discord - Palace
 DiscordPHP\vendor\react\http\src\Io\ReadableBodyStream.php(151): React\Http\Io\
ReadableBodyStream->close() #31 C:\Discord Bots\discord - Palace DiscordPHP\vend
or\react\http\src\Io\ReadableBodyStream.php(33): React\Http\Io\ReadableBodyStrea
m->handleEnd() #32 C:\Discord Bots\discord - Palace DiscordPHP\vendor\evenement\
evenement\src\Evenement\EventEmitterTrait.php(123): React\Http\Io\ReadableBodySt
ream->React\Http\Io\{closure}('{"embed": ["fie...') #33 C:\Discord Bots\discord
- Palace DiscordPHP\vendor\react\http\src\Client\Request.php(163): Evenement\Eve
ntEmitter->emit('data', Array) #34 C:\Discord Bots\discord - Palace DiscordPHP\v
endor\evenement\evenement\src\Evenement\EventEmitterTrait.php(123): React\Http\C
lient\Request->handleData('HTTP/1.1 400 Ba...') #35 C:\Discord Bots\discord - Pa
lace DiscordPHP\vendor\react\stream\src\Util.php(71): Evenement\EventEmitter->em
it('data', Array) #36 C:\Discord Bots\discord - Palace DiscordPHP\vendor\eveneme
nt\evenement\src\Evenement\EventEmitterTrait.php(123): React\Stream\Util::React\
Stream\{closure}('HTTP/1.1 400 Ba...') #37 C:\Discord Bots\discord - Palace Disc
ordPHP\vendor\react\stream\src\DuplexResourceStream.php(193): Evenement\EventEmi
tter->emit('data', Array) #38 C:\Discord Bots\discord - Palace DiscordPHP\vendor
\react\event-loop\src\StreamSelectLoop.php(245): React\Stream\DuplexResourceStre
am->handleData(Resource id #4552) #39 C:\Discord Bots\discord - Palace DiscordPH
P\vendor\react\event-loop\src\StreamSelectLoop.php(212): React\EventLoop\StreamS
electLoop->waitForStreamActivity(387952) #40 C:\Discord Bots\discord - Palace Di
scordPHP\vendor\team-reflex\discord-php\src\Discord\Discord.php(1347): React\Eve
ntLoop\StreamSelectLoop->run() #41 C:\Discord Bots\discord - Palace DiscordPHP\r
un.php(278): Discord\Discord->run() #42 {main} [] []
```

**Comment by valzargaming** [comment]
Attached are the var_dump of the $embed objects being sent, just in case.
[remove_embed.txt](https://github.com/teamreflex/DiscordPHP/files/5653059/remove_embed.txt)
[add_embed.txt](https://github.com/teamreflex/DiscordPHP/files/5653060/add_embed.txt)

**Comment by valzargaming** [comment]
I went back through everything and I'm 99% sure the issue was caused by my $embed objects containing invalid information. In once instance I made a call to ->addField instead of ->addFIeldValues and another instance I entered ->setColor as a string instead of an int. I'm curious as to why neither the ->sendEmbed or ->sendMessage functions picked up on this as I would think there would be some kind of embed validation happening before it was actually sent to Discord.
